BMM.Core/ViewModels/CuratedPlaylistsViewModel.cs

```diff
@@ -1,26 +1,61 @@
 ﻿using System.Collections.Generic;
+using System.Linq;
 using System.Threading.Tasks;
 using BMM.Api.Abstraction;
 using BMM.Api.Implementation.Models;
+using BMM.Core.Extensions;
+using BMM.Core.GuardedActions.Documents.Interfaces;
 using BMM.Core.Implementations.Caching;
-using BMM.Core.Implementations.DocumentFilters;
+using BMM.Core.Implementations.Languages;
+using BMM.Core.Implementations.Security;
 using BMM.Core.ViewModels.Base;
 
 namespace BMM.Core.ViewModels
 {
     public class CuratedPlaylistsViewModel : DocumentsViewModel
     {
+        private readonly IAppLanguageProvider _appLanguageProvider;
+        private readonly IUserStorage _userStorage;
+        private readonly ITranslateDocsAction _translateDocsAction;
+        private readonly IPrepareCoversCarouselItemsAction _prepareCoversCarouselItemsAction;
         public override CacheKeys? CacheKey => CacheKeys.PlaylistGetAll;
 
         public static readonly List<int> FeaturedIds = new List<int> { 31, 32, 33, 35, 36, 37 };
 
-        public CuratedPlaylistsViewModel(IDocumentFilter filter = null) : base(filter)
-        { }
+        public CuratedPlaylistsViewModel(
+            IAppLanguageProvider appLanguageProvider,
+            IUserStorage userStorage,
+            ITranslateDocsAction translateDocsAction,
+            IPrepareCoversCarouselItemsAction prepareCoversCarouselItemsAction)
+        {
+            _appLanguageProvider = appLanguageProvider;
+            _userStorage = userStorage;
+            _translateDocsAction = translateDocsAction;
+            _prepareCoversCarouselItemsAction = prepareCoversCarouselItemsAction;
+        }
 
         public override async Task<IEnumerable<Document>> LoadItems(CachePolicy policy = CachePolicy.UseCacheAndRefreshOutdated)
         {
-            var playlists = await Client.Playlist.GetAll(policy);
-            return playlists;
+            var playlistsDocuments = await Client
+                .Playlist
+                .GetDocuments(_appLanguageProvider.GetAppLanguage(), _userStorage.GetUser().Age, policy);
+
+            playlistsDocuments
+                .Items
+                .OfType<DiscoverSectionHeader>()
+                .ToList()
+                .ForEach(header => header.UseCoverCarousel = true);
+            
+            var translatedItems = await _translateDocsAction.ExecuteGuarded(playlistsDocuments.Items.ToList());
+            var carouselAdjustedItems = await _prepareCoversCarouselItemsAction.ExecuteGuarded(translatedItems);
+
+            carouselAdjustedItems
+                .OfType<DiscoverSectionHeader>()
+                .FirstOrDefault()
+                .IfNotNull(header => header.IsSeparatorVisible = false);
```

Comment:
also, why is this needed?

Reply:
To avoid showing separator directly under the "Playlists" title, which looks bad:

<img width="401" alt="image" src="https://user-images.githubusercontent.com/19508142/181051435-63454a52-6a96-4430-930f-64077fb59209.png">

The same mechanism is used on Home and Browse views.